test/stablex/stablecoin_converter.js

```diff
@@ -467,6 +466,62 @@ contract("StablecoinConverter", async (accounts) => {
       assert.equal((await stablecoinConverter.getBalance.call(user_2, erc20_2)).toString(), basicTrade.deposits[1].amount.sub(getExecutedSellAmount(fullBuyVolumes[1], prices[0], prices[1])).toString(), "Sold tokens were not adjusted correctly")
       assert.equal((await stablecoinConverter.getBalance.call(user_2, feeToken)), fullBuyVolumes[1].toString(), "Bought tokens were not adjusted correctly")
     })
+    it("Ensure fee token is burnt as expected", async () => {
```

Comment:
nit: as expected is not very descriptive as the reader might not now what is expected. Should we say "Ensure half of the fee is burnt"?

Reply:
```suggestion
    it("ensures half of the token imbalance (fees) is burnt", async () => {
```

Reply:
I think, it would be better to split this test case into two separate ones. One where you are testing that fees are burnt, if we hit a new auction, and one where you are testing that no fee's are burnt, if we are still in the same auction

Reply:
Yes, I would agree that there should be two tests, but I don't this test is currently doing both. I will add another.

Reply:
> split this test case into two separate ones 

After splitting this into two test cases, I noticed a substantial amount of duplicated code. It would literally be a copy-paste of the first test (50 lines) followed by this little snippet:

```js
      // Better second solution
      const betterSolution = solutionSubmissionParams(secondTradeExample.solutions[1], accounts, secondOrderIds)
      // This is where the first auction's fees should be burned!
      await stablecoinConverter.submitSolution(
        nextBatchIndex,
        betterSolution.objectiveValue,
        betterSolution.owners,
        betterSolution.touchedOrderIds,
        betterSolution.volumes,
        betterSolution.prices,
        betterSolution.tokenIdsForPrice,
        { from: solver })
      const afterBetterSolutionFeeBalance = await owlProxy.balanceOf(stablecoinConverter.address)
      assert(initialFeeTokenBalance.sub(basicTrade.solutions[0].burntFees).eq(afterBetterSolutionFeeBalance))
```

I would argue that there is no need for a second test just for this last little bit if the first test verifies what we want.